framework/core/Project/Lang.pm

```diff
@@ -42,18 +42,65 @@ use Vcs::Git;
 
 our @ISA = qw(Project);
 my $PID  = "Lang";
+# TODO changeme
+my $RANDOM_TEST_FILE = "$SCRIPT_DIR/build-scripts/$PID/random_tests";
 
 sub new {
-    my $class= shift;
+    my ($class, $work_dir, $commit_db) = @_;
+    $work_dir = $work_dir // "$SCRIPT_DIR/projects";
     my $name = "commons-lang";
     my $src  = "src/main/java";
     my $test = "src/test";
     my $vcs  = Vcs::Git->new($PID,
                              "$REPO_DIR/$name.git",
-                             "$SCRIPT_DIR/projects/$PID/commit-db",
+                             ($commit_db // "$SCRIPT_DIR/projects/$PID/commit-db"),
```

Comment:
Minor: can you move this up to line 50?

Also, shouldn't `$SCRIPT_DIR/projects` be `work_dir`?